etcdserver/auth/auth.go

```diff
@@ -281,11 +285,14 @@ func (s *store) UpdateUser(user User) (User, error) {
 		return old, err
 	}
 
-	hash, err := s.HashPassword(user.Password)
-	if err != nil {
-		return old, err
+	if user.Password != "" {
+		var hash string
+		hash, err = s.HashPassword(user.Password)
```

Comment:
Actually this check was all we need. (remove empty check in HashPassword). 

https://github.com/coreos/etcd/blob/master/etcdserver/api/v2http/client_auth.go#L422-L442 passes user without password (even when user has been created with password), so we need to check this line I think.